tests/components/nsw_fuel_station/test_config_flow.py

```diff
@@ -62,65 +77,176 @@ async def test_form(hass: HomeAssistant, mock_fuelcheckclient: MagicMock) -> Non
+    )
+
+    await hass.async_block_till_done()
 
-    with pytest.raises(vol.Invalid):
-        await hass.config_entries.flow.async_configure(
-            result["flow_id"],
-            user_input={
-                INPUT_FUEL_TYPES: ["P95", "DL"],
-            },
-        )
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "already_configured"
 
 
 @pytest.mark.usefixtures("mock_fuelcheckclient")
 async def test_client_error(
     hass: HomeAssistant, mock_fuelcheckclient: MagicMock
 ) -> None:
-    """Test we get the form."""
+    """Test flow is aborted on client error."""
+    mock_fuelcheckclient.get_fuel_prices.side_effect = FuelCheckError
+    mock_fuelcheckclient.get_reference_data.side_effect = FuelCheckError
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "fetch_failed"
+
+
+@pytest.mark.usefixtures("mock_fuelcheckclient")
+async def test_import_flow_ok(
+    hass: HomeAssistant,
+) -> None:
+    """Test the import configuration flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_IMPORT},
+        data={
+            CONF_STATION_ID: 222,
+            CONF_FUEL_TYPES: ["E10", "DL"],
+        },
+    )
+
+    assert result.get("type") is FlowResultType.CREATE_ENTRY
+    assert result.get("title") == "Joe's Servo"
+    assert result.get("data") == {
+        CONF_STATION_ID: 222,
+        CONF_FUEL_TYPES: ["E10", "DL"],
+    }
+    assert result.get("options") == {}
+
+
+@pytest.mark.usefixtures("mock_fuelcheckclient")
+async def test_import_flow_client_error(
+    hass: HomeAssistant, mock_fuelcheckclient: MagicMock
+) -> None:
+    """Test import flow is aborted on client error."""
+    mock_fuelcheckclient.get_fuel_prices.side_effect = FuelCheckError
     mock_fuelcheckclient.get_reference_data.side_effect = FuelCheckError
 
-    with pytest.raises(ConfigEntryNotReady):
-        await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": config_entries.SOURCE_USER}
-        )
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_IMPORT},
+        data={
+            CONF_STATION_ID: 555,
+            CONF_FUEL_TYPES: ["E10", "DL"],
+        },
+    )
+    assert result["type"] is FlowResultType.ABORT
+
+
+@pytest.mark.usefixtures("mock_fuelcheckclient")
+async def test_import_flow_no_station_found(
+    hass: HomeAssistant,
+) -> None:
+    """Test the import configuration flow, station not found."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_IMPORT},
+        data={
+            CONF_STATION_ID: 555,
+            CONF_FUEL_TYPES: ["E10", "DL"],
+        },
+    )
+
+    assert result.get("type") is FlowResultType.ABORT
```

Comment:
```suggestion
    assert result["type"] is FlowResultType.ABORT
```

This should be optional.